src/window-event-handler.coffee

```diff
@@ -141,8 +141,8 @@ class WindowEventHandler
       @atomEnvironment.hide()
     @reloadRequested = false
 
-    @atomEnvironment.storeDefaultWindowDimensions()
     @atomEnvironment.storeWindowDimensions()
+    @atomEnvironment.saveState()
```

Comment:
Now we fire off one last async save before unloading the window. @kuychaco had chosen not to bother with this initially, since we're already saving periodically, and it was unclear whether the async save would complete before the unload. I think it does work though, and it's important to ensure we store the latest window dimensions.